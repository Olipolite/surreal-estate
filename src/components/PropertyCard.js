import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faBed, faBathtub } from "@fortawesome/free-solid-svg-icons";
import Proptypes from "prop-types";
import "../styles/propertycard.css";

const PropertyCard = ({
  title,
  type,
  bathrooms,
  bedrooms,
  price,
  city,
  email,
}) => {
  return (
    <div className="property-card">
      <div className="title">{title}</div>
      <div className="type">{type}</div>
      <div className="bathrooms">
        <FontAwesomeIcon icon={faBathtub} />
        {bathrooms}
      </div>
      <div className="bedrooms">
        <FontAwesomeIcon icon={faBed} />
        {bedrooms}
      </div>
      <div className="price">{price}</div>
      <div className="city">{city}</div>
      <div className="email">{email}</div>
      <div>{email}</div>
    </div>
  );
};

export default PropertyCard;

PropertyCard.propTypes = {
  title: Proptypes.string,
  type: Proptypes.string,
  bathrooms: Proptypes,
  bedrooms: Proptypes,
  price: Proptypes,
  city: Proptypes,
  email: Proptypes,
}.isRequired;
