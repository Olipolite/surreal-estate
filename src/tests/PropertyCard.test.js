import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "test title",
    type: "test type",
    bathrooms: 1,
    bedrooms: 1,
    price: 9999,
    city: "test city",
    email: "test email",
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
