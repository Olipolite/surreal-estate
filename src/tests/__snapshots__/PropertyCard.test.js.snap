// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="title"
    >
      test title
    </div>
    <div
      class="type"
    >
      test type
    </div>
    <div
      class="bathrooms"
    >
      1
    </div>
    <div
      class="bedrooms"
    >
      1
    </div>
    <div
      class="price"
    >
      9999
    </div>
    <div
      class="city"
    >
      test city
    </div>
    <div
      class="email"
    >
      test email
    </div>
    <div>
      test email
    </div>
  </div>
</DocumentFragment>
`;
